.isort.cfg

```diff
@@ -1,7 +1,7 @@
 [settings]
 
 line_length=100
-sections=FUTURE,STDLIB,THIRDPARTY,LOCALFOLDER
+sections=FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
```

Comment:
this change should not be necessary

Reply:
Yeah I was reluctant to change project config as part of an unrelated PR, but I explain in the commit why it seems necessary.

Here's a lot more detail:

If I take a fresh checkout of Dagster and hit "save" in vscode[1] on a Python file, I get the following error:

> [ERROR 2022-1-23 15:25:57.310]:     raise MissingSection(import_module=import_from, section=placed_module)
isort.exceptions.MissingSection: Found dagster.core.instance import while parsing, but FIRSTPARTY was not included in the `sections` setting of your config. Please add it before continuing
See https://pycqa.github.io/isort/#custom-sections-and-ordering for more info.

The isort documentation says:

> You can change the section order with sections option from the default of:
>
>     FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
>
> to your preference (if defined, omitting a default section may cause errors)...

(Note that this commit sets the config to the above default setting for isort)

Upon further research I discovered that Dagster requires isort version 4. If I upgrade to isort 5 within the project and run it from the commandline I get the above error.

So:

1. that addition indeed isn't necessary now, but will be whenever you upgrade to isort 5
2. it's broken by default currently for vscode users, as the vscode Python extension bundles isort 5
3. best I can tell, the addition of `FIRSTPARTY` to the isort config doesn't change behavior in version 4, but allows it to work for vscode users and isort 5

I did find an example file (`python_modules/dagster/dagster/cli/pipeline.py`) that behaves differently with that setting under isort 5 than under isort 4. Currently the imports in that file look like:

```python
import click
import pendulum
import yaml
from dagster import PipelineDefinition
...
from dagster.utils.interrupts import capture_interrupts
from tabulate import tabulate

from .config_scaffolder import scaffold_pipeline_config
```

isort 4 with that change still sorts the imports the same. Under isort 5, with the (required) addition of `FIRSTPARTY` sorts it as follows:

```python
import click
import pendulum
import yaml
from tabulate import tabulate

from dagster import PipelineDefinition
...
from dagster.utils.interrupts import capture_interrupts

from .config_scaffolder import scaffold_pipeline_config
```

That's actually *correct* because `tabulate` is a third-party module. isort 4 is sorting imports incorrectly!

I'll back out that change to the isort config, but I recommend updating to isort 5.

[1] With the Python extension and "organize imports on save" in settings.json:

```json
    "[python]": {
        "editor.codeActionsOnSave": {
            "source.organizeImports": true
        }
    },
```

Reply:
Really appreciate the detailed feedback here -- if make dev_install has been run in the current environment, this issue does not occur (because, as you note, the correct version of isort will be installed) so this is a rare-ish error to hit, but you're right that there is not a particularly compelling reason to keep this behavior this way.

I've added this change to a separate PR to keep things organized: https://github.com/dagster-io/dagster/pull/6788

Once again, thanks for your contribution -- really appreciate how thorough you've been on this!

Reply:
> if make dev_install has been run in the current environment

FWIW I did follow the contributing instructions and run `make dev_install`, so I did have isort 4 in my virtualenv. So isort (4) would work from the command line, but organizing imports on save would still fail in vscode because the Python extension bundles isort 5.